lib/gollum/views/helpers.rb

```diff
@@ -53,8 +53,15 @@ def self.included(base)
         end
       end
 
-      def page_route(page)
-        "#{base_url}/#{page}".gsub(/\/{2,}/, '/') # remove double slashes
+      def page_route(page = nil)
+        clean_url(@base_url, page)
+      end
+
+      def clean_url(*url)
+        url.compact!
```

Comment:
Why is this `compact!` call necessary -- and if it is, shouldn't it happen after the `nil` check on the next line?

Reply:
Note that `url` is an array that receives variable-length arguments. For example, `clean_url(nil)` and `clean_url("base", nil)` result in `url = [nil]` and `url = ["base", nil]`, respectively. So `compact!` is enough, and the next line `return url if url.nil?` should be `return url if url.empty?` instead.